File: app/src/main/java/shy/car/sdk/travel/user/data/UserBase.java
package shy.car.sdk.travel.user.data;


import android.databinding.BaseObservable;
import android.databinding.Bindable;

import com.google.gson.annotations.SerializedName;

import shy.car.sdk.BR;


/**
 * Created by LZ on 2016/8/26.
 * 用户
 */
public class UserBase extends BaseObservable {
    public static class PromissState {
        public static final int MONEY_PAYED = 1;
        public static final int MONEY_NOT_PAYED = 2;
    }


    public static final String UID = "uid";
    public static final String PHONE = "phone";
    public static final String ACCESS_TOKEN = "access_token";
    public static final String AVATAR = "avatar";
    public static final String NICKNAME = "name";
    public static final String EXPIRES_IN = "expires_in";
    public static final String BUYER_TYPE = "type";
    public static final String PASSWORD = "password";
    public static final String STATUS = "status";

    /**
     * uid : 10314
     * phone : 18577845220
     * bind_weixin : 0
     */

    @SerializedName( UID)
    private int uid;
    @SerializedName( PHONE)
    private String phone;
    //    @SerializedName( BIND_WEIXIN)
//    private int bindWeixin;
    @SerializedName( ACCESS_TOKEN)
    private String access_token;
    @SerializedName( NICKNAME)
    private String nickName;
    @SerializedName( AVATAR)
    private String avatar;
    @SerializedName( EXPIRES_IN)
    private long expiresIn;
    /**
     * 1:已认证。2：未认证
     */
    @SerializedName( STATUS)
    private int status;

    /**
     * 密码：1:已设置，0：未设置密码
     */
    @SerializedName( PASSWORD)
    private int password;

    /**
     * 支付密码 1 已设置 0 未设置
     */
    @SerializedName( "payment_password")
    private int paymentPassword;
    /**
     * 推送设置 1 接收 2 不接收
     */
    @SerializedName( "msg_remind")
    private int msgRemind;

    /**
     * 保证金缴纳.>0已缴纳，<=0未缴纳
     */
    @SerializedName( "promise_money")
    private double promiseMoney;

    /**
     * 用户实名认证
     * .1=已认证，0=未认证
     */
    @SerializedName( "user_verify")
    private int userVerify;

    public boolean isUserVerify() {
        return getUserVerify() == 1;
    }

    @Bindable
    public int getUserVerify() {
        return userVerify;
    }

    public void setUserVerify(int userVerify) {
        this.userVerify = userVerify;
        notifyPropertyChanged(BR.userVerify);
    }

    /**
     * 保证金缴纳：1：已缴纳，2：未缴纳
     */
    @Bindable
    public boolean getPromiseMoneyState() {
        return promiseMoney > 0;
    }


    @Bindable
    public double getPromiseMoney() {
        return promiseMoney;
    }

    public void setPromiseMoney(double promiseMoney) {
        this.promiseMoney = promiseMoney;
        notifyPropertyChanged(BR.promiseMoney);
    }

    @Bindable
    public int getMsgRemind() {
        return msgRemind;
    }

    public void setMsgRemind(int msgRemind) {
        this.msgRemind = msgRemind;
        notifyPropertyChanged(BR.msgRemind);
    }

    public int getPaymentPassword() {
        return paymentPassword;
    }

    public void setPaymentPassword(int paymentPassword) {
        this.paymentPassword = paymentPassword;
    }

    @Bindable
    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
        notifyPropertyChanged(BR.status);
    }

    @Bindable
    public int getPassword() {
        return password;
    }

    public void setPassword(int password) {
        this.password = password;
        notifyPropertyChanged(BR.password);
    }

    private int loginType;

    private long loginTime;

    public void setLoginTime(long loginTime) {
        this.loginTime = loginTime;
    }

    public long getLoginTime() {
        return loginTime;
    }

    public int getLoginType() {
        return loginType;
    }

    public void setLoginType(int loginType) {
        this.loginType = loginType;
    }

    @Bindable
    public int getUid() {
        return uid;
    }

    public void setUid(int uid) {
        this.uid = uid;
        notifyPropertyChanged(BR.uid);
    }

    @Bindable
    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        if ("0".equals(phone)) {
            this.phone = "";
        } else {
            this.phone = phone;
        }
        notifyPropertyChanged(BR.phone);
    }

    public String getAccess_token() {
        return access_token;
    }

    public void setAccess_token(String access_token) {
        this.access_token = access_token;
        notifyPropertyChanged(BR.login);
    }

    @Bindable
    public String getNickName() {
        return nickName;
    }

    public void setNickName(String nickName) {
        this.nickName = nickName;
        notifyPropertyChanged(BR.nickName);
    }

    @Bindable
    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
        notifyPropertyChanged(BR.avatar);
    }

    public long getExpiresIn() {
        return expiresIn;
    }

    public void setExpiresIn(long expiresIn) {
        this.expiresIn = expiresIn;
    }
    
    @Bindable
    public boolean isLogin() {
        return com.base.util.StringUtils.isNotEmpty(access_token);
    }

    @Override
    public String toString() {
        return "UserBase{" +
                "uid=" + uid +
                ", phone='" + phone + '\'' +
//               ", bindWeixin=" + bindWeixin +
                ", access_token='" + access_token + '\'' +
                ", nickName='" + nickName + '\'' +
                ", avatar='" + avatar + '\'' +
                ", expiresIn=" + expiresIn +
                ", loginType=" + loginType +
                ", loginTime=" + loginTime +
                '}';
    }
}

File: app/src/main/java/shy/car/sdk/travel/user/data/UserBase.kt
package shy.car.sdk.travel.user.data


import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import com.base.util.StringUtils

import com.google.gson.annotations.SerializedName

import shy.car.sdk.BR
import shy.car.sdk.app.constant.ParamsConstant.City


/**
 * Created by LZ on 2016/8/26.
 * 用户
 */
open class UserBase : BaseObservable() {

    /**
     * uid : 10314
     * phone : 18577845220
     * bind_weixin : 0
     */
    @SerializedName(PHONE)
    @get:Bindable
    var phone: String = ""
        set(phone) {
            if ("0" == phone) {
                field = ""
            } else {
                field = phone
            }
            notifyPropertyChanged(BR.phone)
        }

    @SerializedName(ACCESS_TOKEN)
    var accessToken: String = ""

    @SerializedName(REFRESH_TOKEN)
    var refreshToken: String = ""


    @SerializedName("max_cancel_num")
    @get:Bindable
    var maxCancelNum: Int = 5
        set(value) {
            field = value
            notifyPropertyChanged(BR.maxCancelNum)
        }
    @SerializedName(Name)
    @get:Bindable
    var name: String = ""
        set(value) {
            field = value
            notifyPropertyChanged(BR.name)
        }

    @SerializedName(AVATAR)
    @get:Bindable
    var avatar: String = ""
        set(avatar) {
            field = avatar
            notifyPropertyChanged(BR.avatar)
        }
    @SerializedName(Birthday)
    @get:Bindable
    var birthday: String = ""
        set(birthday) {
            field = birthday
            notifyPropertyChanged(BR.birthday)
        }
    /**
     * 性别（0保密 1男 2女）
     */
    @SerializedName(SEX)
    @get:Bindable
    var sex: Int = 0
        set(sex) {
            field = sex
            notifyPropertyChanged(BR.sex)
        }
    /**
     * 用户等级
     */
    @SerializedName("rank")
    @get:Bindable
    var rank: Int = 0
        set(rank) {
            field = rank
            notifyPropertyChanged(BR.rank)
        }

    /**
     * 用户等级名称
     */
    @SerializedName("rank_text")
    @get:Bindable
    var rankText: String = ""
        set(value) {
            field = value
            notifyPropertyChanged(BR.rankText)
        }
    /**
     * 昵称
     */
    @SerializedName("nickname")
    @get:Bindable
    var nickName: String = ""
        set(nickName) {
            field = nickName
            notifyPropertyChanged(BR.nickName)
        }
    /**
     * 用户类型
     */
    @SerializedName("type")
    @get:Bindable
    var type: Int = 0
        set(type) {
            field = type
            notifyPropertyChanged(BR.type)
        }

    /**
     * 类型名称
     */
    @SerializedName("type_text")
    @get:Bindable
    var typeText: String = "个人用户"
        set(type_text) {
            field = type_text
            notifyPropertyChanged(BR.typeText)
        }

    /**
     * 认证状态（0未认证 1认证中 2已认证）
     */
    @SerializedName("is_identity_auth")
    @get:Bindable
    var identityAuth: Int = 0
        set(value) {
            field = value
            notifyPropertyChanged(BR.identityAuth)
            notifyPropertyChanged(BR.isIdentityAuth)
            notifyPropertyChanged(BR.isDeposit)
        }
    /**
     * 认证状态（0未认证 1认证中 2已认证）
     */
    @SerializedName("is_driver_auth")
    @get:Bindable
    var driverAuth: Int = 0
        set(value) {
            field = value
            notifyPropertyChanged(BR.isDriverAuth)
            notifyPropertyChanged(BR.driverAuth)
        }

    @Bindable
    fun getIsDriverAuth(): Boolean {
        return driverAuth == 2
    }

    @Bindable
    fun getIsIdentityAuth(): Boolean {
        return identityAuth == UserState.UserIdentityAuth.Identited
    }

    /**
     * 保证金
     */
    @SerializedName("deposit")
    @get:Bindable
    var deposit: Double = 0.0
        set(value) {
            field = value
            notifyPropertyChanged(BR.deposit)
            notifyPropertyChanged(BR.isIdentityAuth)
            notifyPropertyChanged(BR.isDeposit)
        }

    /**
     * 保证金（0未缴 1已缴）
     */
    @SerializedName("is_deposit")

    var isDeposited: Int = 0
        set(value) {
            field = value
            notifyPropertyChanged(BR.deposit)
            notifyPropertyChanged(BR.isIdentityAuth)
            notifyPropertyChanged(BR.isDeposit)
        }

    /**
     * 保证金（0未缴 1已缴）
     */
    @Bindable
    fun getIsDeposit(): Boolean {
        return isDeposited == 1
    }


    /**
     * 账户余额
     */
    @SerializedName("balance")
    @get:Bindable
    var balance: Double = 0.0
        set(balance) {
            field = balance
            notifyPropertyChanged(BR.balance)
        }
    /**
     * 用户积分
     */
    @SerializedName("score")
    @get:Bindable
    var score: Int = 0
        set(score) {
            field = score
            notifyPropertyChanged(BR.score)
        }
    /**
     * 银行卡数量
     */
    @SerializedName(BankCardNum)
    @get:Bindable
    var bankCardNum: Int = 0
        set(bank_card_num) {
            field = bank_card_num
            notifyPropertyChanged(BR.bankCardNum)
        }
    /**
     * 优惠劵数量
     */
    @SerializedName(CouponNum)
    @get:Bindable
    var couponNum: Int = 0
        set(coupon_num) {
            field = coupon_num
            notifyPropertyChanged(BR.couponNum)
        }


    @SerializedName(EXPIRES_IN)
    var expiresIn: Long = 0

    @SerializedName(Scope)
    @get:Bindable
    var scope: String = ""
        set(value) {
            field = value
            notifyPropertyChanged(BR.scope
            )
        }
    @SerializedName(Profession)
    @get:Bindable
    var profession: String? = null
        set(profession) {
            field = profession
            notifyPropertyChanged(BR.profession)
        }
    @SerializedName(City)
    @get:Bindable
    var city: String? = null
        set(city) {
            field = city
            notifyPropertyChanged(BR.city)
        }

    @SerializedName(TokenType)
    var tokenType: String = ""


    var loginTime: Long = 0

    @get:Bindable
    var login: Boolean = false
        set(value) {
            field = value
            notifyPropertyChanged(BR.login)
        }
        get() {
            return StringUtils.isNotEmpty(accessToken)
        }


    companion object {
        const val PHONE = "phone"
        const val ACCESS_TOKEN = "access_token"
        const val REFRESH_TOKEN = "refresh_token"
        const val AVATAR = "avatar"
        const val Name = "name"
        const val EXPIRES_IN = "expires_in"
        const val TokenType = "token_type"
        const val Scope = "scope"
        const val SEX = "sex"
        const val CouponNum = "coupon_num"
        const val BankCardNum = "bank_card_num"
        const val Profession = "profession"
        const val Birthday = "birthday"
    }

}
